es72x/src/main/scala/tech/beshu/ror/es/utils/ThreadRepo.scala

```diff
@@ -0,0 +1,48 @@
+/*
+ *    This file is part of ReadonlyREST.
+ *
+ *    ReadonlyREST is free software: you can redistribute it and/or modify
+ *    it under the terms of the GNU General Public License as published by
+ *    the Free Software Foundation, either version 3 of the License, or
+ *    (at your option) any later version.
+ *
+ *    ReadonlyREST is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *    GNU General Public License for more details.
+ *
+ *    You should have received a copy of the GNU General Public License
+ *    along with ReadonlyREST.  If not, see http://www.gnu.org/licenses/
+ */
+package tech.beshu.ror.es.utils
+
+import org.elasticsearch.rest.RestChannel
+import tech.beshu.ror.SecurityPermissionException
+
+object ThreadRepo {
+  private val threadLocalChannel = new ThreadLocal[RestChannel]
+
+  def setRestChannel(restChannel: RestChannel): Unit = {
+    threadLocalChannel.set(restChannel)
+  }
+
+  def getRestChannel: Option[RestChannel] = {
+    val channel = threadLocalChannel.get
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Usage of get on optional type.](https://app.codacy.com/manual/scarduzio/elasticsearch-readonlyrest-plugin/pullRequest?prid=4803411)